http-server/src/main/java/com/facebook/airlift/http/server/HttpServer.java

```diff
@@ -427,9 +430,19 @@ private static ServletContextHandler createServletContext(
         context.addServlet(servletHolder, "/*");
 
         for (Map.Entry<String, Servlet> servlet : servlets.entrySet()) {
-            ServletHolder holder = new ServletHolder(servlet.getValue());
-            holder.setInitParameters(ImmutableMap.copyOf(parameters));
-            context.addServlet(holder, servlet.getKey());
+            if (config.isAuthorizationEnabled()) {
+                SecureServlet secureServlet = new SecureServlet(servlet.getValue(),
```

Comment:
It looks like we decided to go with a config. Please add a `checkArgument(authorizer != null, "to enable authorization authorizer implementation must be provided")`. Feel free to change the message if you think doesn't sound good enough.